src/sdk/index.ts

```diff
@@ -417,38 +330,95 @@ export async function createPlayground(
-  const configParam =
-    typeof config === 'string'
-      ? { config }
-      : config && typeof config === 'object' && Object.keys(config).length
-        ? { x: 'code/' + compressToEncodedURIComponent(JSON.stringify(config)) }
-        : {};
-
-  let encodedParams;
-  if (params && typeof params === 'object') {
+  const {
+    appUrl = 'https://livecodes.io',
+    params = {},
+    config = {},
+    headless,
+    import: importId,
+    lite,
+    view,
+    ...otherOptions
+  } = options;
+
+  const playgroundUrl = new URL(appUrl);
+  const hashParams = new URLSearchParams();
+
+  // Add other options to search params
+  Object.entries(otherOptions).forEach(([key, value]) => {
+    if (value !== undefined) {
+      playgroundUrl.searchParams.set(key, String(value));
+    }
+  });
+
+  const isHeadless = options.view === 'headless' || headless; // for backwards compatibility;
+
+  if (lite) {
+    // eslint-disable-next-line no-console
+    console.warn(
+      `Deprecation notice: "lite" option is deprecated. Use "config: { mode: 'lite' }" instead.`,
+    );
+    if (typeof config === 'object' && config.mode == null) {
+      config.mode = 'lite';
+    } else {
+      playgroundUrl.searchParams.set('lite', 'true');
+    }
+  }
+
+  if (view) {
+    // eslint-disable-next-line no-console
+    console.warn(
+      `Deprecation notice: The "view" option has been moved to "config.view". For headless mode use "headless: true".`,
+    );
+    if (typeof config === 'object' && config.view == null && view !== 'headless') {
+      config.view = view;
+    } else {
+      playgroundUrl.searchParams.set('view', view);
+    }
+  }
+
+  if (typeof config === 'string') {
+    try {
+      new URL(config);
+      playgroundUrl.searchParams.set('config', encodeURIComponent(config));
+    } catch {
+      throw new Error(`"config" is not a valid URL or configuration object.`);
+    }
+  } else if (config && typeof config === 'object' && Object.keys(config).length > 0) {
+    if (config.title && config.title !== 'Untitled Project') {
+      playgroundUrl.searchParams.set('title', config.title);
+    }
+    if (config.description && config.description.length > 0) {
+      playgroundUrl.searchParams.set('description', config.description);
+    }
+    hashParams.set('config', 'code/' + compressToEncodedURIComponent(JSON.stringify(config)));
+  }
+
+  // handle params
+  if (params && typeof params === 'object' && Object.keys(params).length > 0) {
     try {
-      encodedParams = compressToEncodedURIComponent(JSON.stringify(params));
+      hashParams.set('params', compressToEncodedURIComponent(JSON.stringify(params)));
     } catch {
       (Object.keys(params) as Array<keyof UrlQueryParams>).forEach((param) => {
-        (params as any)[param] = encodeURIComponent(String(params[param]));
+        playgroundUrl.searchParams.set(param, encodeURIComponent(String(params[param]))); // TODO: should this be in hash as well? what kind of error is thrown here
```

Comment:
This is already a backward-compatibility thing.
We can just keep it as it is.
`JSON.stringify` can [throw errors](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#exceptions).